src/netbase.h

```diff
@@ -37,6 +41,91 @@ class proxyType
     bool randomize_credentials;
 };
 
+struct timeval MillisToTimeval(int64_t nTimeout);
+
+// Represents a read/write socket.
+class Socket
+{
+public:
+    virtual ssize_t send(const char* buf, size_t len, int flags) = 0;
+
+    virtual ssize_t recv(char* buf, size_t len, int flags) = 0;
+
+#ifdef USE_POLL
+    virtual int poll(short events, int timeout_ms) = 0;
+#endif
+
+    virtual int select(bool watch_read, bool watch_write, int timeout_ms) = 0;
+
+    virtual bool IsSelectable() const = 0;
+};
+
+// Represents a non-owned SOCKET.
+//
+// The underlying SOCKET will NOT be closed automatically on object destruction.
+class SocketImpl : public Socket
```

Comment:
Just would like to point out that this is currently unused in the context of this fuzz test. Is this intentional?

Reply:
It is used in `netbase.cpp`? :)

```
$ git grep SocketImpl
src/netbase.cpp:        SocketImpl socket{hSocket};
src/netbase.cpp:        SocketImpl socket{hSocket};
src/netbase.h:class SocketImpl : public Socket
src/netbase.h:    explicit SocketImpl(SOCKET socket) : m_socket{socket}
```

Reply:
I meant that even though it's used, the coverage isn't updated for `SocketImpl` functions

Reply:
I think that is expected: it is currently only from `ConnectThroughProxy` which is not fuzzed :)

Reply:
Ah derp! I mistakenly thought it was fuzzed for some reason, but `FuzzedSocket` is the one that's used. Is the idea that they be used together in some manner eventually?

Reply:
@Crypt-iQ It is only used as a way to wrap an existing `SOCKET`  as a `Socket` instance: no further plans beyond that :)